### DISCUSSION TITLE
Accessing a field with arguments in cache.readFragment

### DISCUSSION THREAD

**Discussion by vitosamson** [urql-graphql/urql#1147] [Q&A] [answered]
I'm trying to do something like the following inside an optimistic update:

```ts
modifyCapabilities({ userId, groupId }, cache) {
  const user = cache.readFragment(
    gql`
      fragment _ on UserQuery {
        id
        capabilities(byGroup: $groupId)
      }
    `,
    { id: userId, groupId, __typename: 'UserQuery' }
  )

  return {
    // ... modified user object
  }
}
```

where `capabilities` is just a `[String!]!` type.

Whenever I include the `capabilities` field in the fragment, `readFragment` returns null. If I remove that field, the `readFragment` successfully returns that user. The field definitely exists in the cache, according to the urql dev tools.

Is it not possible to query a field with variables like that in `readFragment`?

**Comment by JoviDeCroock** [comment, marked as answer]
Have you tried passing the variable to the thirth argument, the second is meant to be a describing entity and the thirth one is used to signal variables https://github.com/FormidableLabs/urql/blob/main/exchanges/graphcache/src/store/store.ts#L191

**Reply by vitosamson** [reply]
That did it, thanks!

**Reply by vitosamson** [reply]
Where do the docs live? I'd like to open a PR to include that information.

**Reply by JoviDeCroock** [reply]
In here: https://github.com/FormidableLabs/urql/tree/main/docs/graphcache
